test/fft_cpp17_allocator.cpp

```diff
@@ -0,0 +1,126 @@
+///////////////////////////////////////////////////////////////////
+//  Copyright Eduardo Quintana 2021
+//  Copyright Janek Kozicki 2021
+//  Copyright Christopher Kormanyos 2021
+//  Distributed under the Boost Software License,
+//  Version 1.0. (See accompanying file LICENSE_1_0.txt
+//  or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+int global_error_count{0};
+
+#if __cplusplus >= 201700
+
+#include <cstdlib>
+#include <boost/math/fft/bsl_backend.hpp>
+#include <memory_resource>
+#include <array>
+
+
+bool new_is_on{true};
+
+void * operator new(size_t size)
+{   
+  if(new_is_on==false)
+    throw std::bad_alloc{};
+  void * p = std::malloc(size);
+  return p;
+}
+void * operator new[](size_t size)
+{   
+  if(new_is_on==false)
+   throw std::bad_alloc{};
+  void * p = std::malloc(size);
+  return p;
+}
+void * operator new(size_t size, size_t align)
+{   
+  if(new_is_on==false)
+    throw std::bad_alloc{};
+  void * p = std::aligned_alloc(size,align);
+  return p;
+}
+void * operator new[](size_t size, size_t align)
+{   
+  if(new_is_on==false)
+   throw std::bad_alloc{};
+  void * p = std::aligned_alloc(size,align);
+  return p;
+}
+
+using namespace boost::math::fft;
+
+template<class Backend>
+void test_inverse(int N, int tolerance)
+{
+  std::array<std::byte,200000> buf;
+  std::pmr::monotonic_buffer_resource
```

Comment:
with the C++17 polymorphic memory resource (pmr) feature, creating custom allocators is very simple. Our custom allocator interface integrates with pmr out-of-the-box.